src/server/src/main/java/io/cassandrareaper/service/RepairScheduleService.java

```diff
@@ -117,6 +123,71 @@ public RepairSchedule storeNewRepairSchedule(
         .adaptive(adaptive)
         .percentUnrepairedThreshold(percentUnrepairedThreshold);
 
-    return context.storage.addRepairSchedule(scheduleBuilder);
+    RepairSchedule repairSchedule = context.storage.addRepairSchedule(scheduleBuilder);
+    registerScheduleMetrics(repairSchedule.getId());
+    return repairSchedule;
+  }
+
+  public void deleteRepairSchedule(UUID repairScheduleId) {
+    unregisterScheduleMetrics(repairScheduleId);
+    context.storage.deleteRepairSchedule(repairScheduleId);
+  }
+
+  private void registerRepairScheduleMetrics(Collection<RepairSchedule> allRepairSchedules) {
+    allRepairSchedules.forEach(schedule -> registerScheduleMetrics(schedule.getId()));
+  }
+
+
+  private void registerScheduleMetrics(UUID repairScheduleId) {
+    RepairSchedule schedule = context.storage.getRepairSchedule(repairScheduleId).get();
+    RepairUnit repairUnit = context.storage.getRepairUnit(schedule.getRepairUnitId());
+    String metricName = metricName("millisSinceLastRepairForSchedule",
+            repairUnit.getClusterName(),
+            repairUnit.getKeyspaceName(),
+            schedule.getId());
+
+    if (!context.metricRegistry.getMetrics().containsKey(metricName)) {
+      context.metricRegistry.register(metricName, getMillisSinceLastRepairForSchedule(schedule.getId()));
+    }
+  }
+
+  private void unregisterScheduleMetrics(UUID repairScheduleId) {
+    RepairSchedule schedule = context.storage.getRepairSchedule(repairScheduleId).get();
+    RepairUnit repairUnit = context.storage.getRepairUnit(schedule.getRepairUnitId());
+    String metricName = metricName("millisSinceLastRepairForSchedule",
+        repairUnit.getClusterName(),
+        repairUnit.getKeyspaceName(),
+        schedule.getId());
+
+    if (context.metricRegistry.getMetrics().containsKey(metricName)) {
+      context.metricRegistry.remove(metricName);
+    }
+  }
+
+  private Gauge<Long> getMillisSinceLastRepairForSchedule(UUID repairSchedule) {
+    return () -> {
+      Optional<RepairSchedule> schedule = context.storage.getRepairSchedule(repairSchedule);
+
+      Long lastRepairDate = schedule.orElseThrow(() -> new IllegalArgumentException("Repair schedule not found"))
+          .getRunHistory().stream()
+          .map(uuid -> context.storage.getRepairRun(uuid))
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .map(RepairRun::getEndTime)
+          .filter(Objects::nonNull)
+          .max(DateTime::compareTo)
+          .map(dateTime -> DateTime.now().getMillis() - dateTime.getMillis())
+          .orElse(DateTime.now().getMillis()); // Return epoch if no repairs from this schedule ever completed
+      return lastRepairDate;
```

Comment:
Shouldn't `lastRepairDate` be renamed to `millisSinceLastRepair` for example?

Reply:
Yes, I'll rename the variable.